presto-main/src/main/java/com/facebook/presto/sql/analyzer/Analysis.java

```diff
@@ -103,6 +105,17 @@ public void setQuery(Query query)
         this.query = query;
     }
 
+    public Optional<Explain> getExplainAnalyze()
+    {
+        return explainAnalyze;
+    }
+
+    public void setExplainAnalyze(Explain explainAnalyze)
```

Comment:
Why do we need a specialized method to set this? The analysis object is meant to be a "query + annotations" structure. If we're expanding to support non-queries, we should change the `query` field to be a Statement.